charms/knative-serving/src/manifests/KnativeServing.yaml.j2

```diff
@@ -44,3 +44,16 @@ spec:
       {{ container }}: {{ image }}
       {% endfor %}
 {% endif %}
+{% if http_proxy and https_proxy and no_proxy %}
```

Comment:
I had a hard time deciding: should users be able to configure only one of those envs? or they have to set all of them before they take effect in the deployment? if we choose the latter probably we should document it well in the description

Reply:
I would say that the users should be able to define any of these and not all of them. This way, the charm's configuration values will represent what actually exists in the knativeServing CR.

Reply:
Thus, we 'll need to have an `if` for each config separately

Reply:
and probably keep this one as well but with `or` instead of `and`?

Reply:
good point, updated